File: apps/android/app/src/qa/java/chat/rogi/rogichat/preview/NotificationPreview.kt
package chat.rogi.rogichat.preview

import androidx.compose.foundation.layout.Box
import androidx.compose.material3.*
import androidx.compose.runtime.*
import chat.rogi.rogichat.feature.settings.*

private enum class NotificationExample(val label: String) {
    DEVICE("실제 기기"), UNKNOWN("미확인 예시"), READING("조회 중 예시"), DENIED("차단 예시"),
    ALLOWED("허용 예시"), NO_CHANNELS("채널 없음 예시"), BLOCKED_CHANNELS("채널 차단 예시"), FAILED("조회 오류 예시")
}

@Composable
fun NotificationPreview() {
    var example by remember { mutableStateOf(NotificationExample.DEVICE) }
    var expanded by remember { mutableStateOf(false) }
    Box {
        TextButton(onClick = { expanded = true }) { Text("알림 화면: ${example.label}") }
        DropdownMenu(expanded, onDismissRequest = { expanded = false }) {
            NotificationExample.entries.forEach { value ->
                DropdownMenuItem(text = { Text(value.label) }, onClick = { example = value; expanded = false })
            }
        }
    }
    key(example) {
        if (example == NotificationExample.DEVICE) NotificationSettingsScreen()
        else {
            Text("OS 상태 예시 · 실제 기기 설정과 달라요. 권한 요청이나 설정 변경은 실행하지 않아요.")
            val started = NotificationReadState().begin()
            val state = when (example) {
                NotificationExample.UNKNOWN -> NotificationReadState()
                NotificationExample.READING -> started
                NotificationExample.FAILED -> started.fail(started.revision)
                else -> started.finish(started.revision, NotificationSnapshot(
                    if (example == NotificationExample.DENIED) NotificationAuthorization.DENIED else NotificationAuthorization.ALLOWED,
                    blockedChannels = if (example == NotificationExample.BLOCKED_CHANNELS) 2 else 0,
                    hasChannels = example != NotificationExample.NO_CHANNELS))
            }
            NotificationSettingsContent(state)
        }
    }
}

File: apps/ios/Sources/QA/NotificationPreview.swift
#if ROGICHAT_QA
import SwiftUI

private enum NotificationExample: String, CaseIterable {
    case device = "실제 기기", unknown = "미확인 예시", reading = "조회 중 예시", notRequested = "미요청 예시"
    case denied = "차단 예시", allowed = "허용 예시", alertsOff = "표시 꺼짐 예시", quiet = "조용한 알림 예시", failed = "조회 오류 예시"
}
struct NotificationPreview: View {
    @State private var example: NotificationExample = .device
    private var state: NotificationReadState {
        var state = NotificationReadState()
        if example == .unknown { return state }
        let ticket = state.begin()
        if example == .reading { return state }
        if example == .failed { state.fail(ticket); return state }
        let authorization: NotificationAuthorization
        switch example {
        case .notRequested: authorization = .notRequested
        case .denied: authorization = .denied
        case .quiet: authorization = .quiet
        default: authorization = .allowed
        }
        state.finish(ticket, NotificationSnapshot(authorization: authorization, alertsEnabled: example != .alertsOff))
        return state
    }
    var body: some View {
        Picker("알림 화면", selection: $example) {
            ForEach(NotificationExample.allCases, id: \.self) { Text($0.rawValue).tag($0) }
        }.pickerStyle(.menu)
        Group {
            if example == .device { NotificationSettingsScreen() }
            else {
                Text("OS 상태 예시 · 실제 기기 설정과 달라요. 권한 요청이나 설정 변경은 실행하지 않아요.").font(.footnote)
                NotificationSettingsContent(state: state)
            }
        }.id(example)
    }
}
#endif
